lib/rules/no-underscore-dangle.js

```diff
@@ -125,6 +130,30 @@ module.exports = {
                 node.object.object.type === "ThisExpression";
         }
 
+        /**
+         * Check if function parameter has a underscore at the beginning.
```

Comment:
This description isn't quite correct. It's a bit confusing, but `hasTrailingUnderscore` actually checks for both leading and trailing underscores. Would you be willing to rename all these methods from `checkForTrailingUnderscore*` to `checkForDanglingUnderscore*`? If not, we can always follow up in a separate PR.